lib/cursor/QueryCursor.js

```diff
@@ -194,6 +194,21 @@ QueryCursor.prototype.close = function(callback) {
   }, this.model.events);
 };
 
+/**
+ * Rewind this cursor to its uninitialized state. Any options that are present on the cursor will
+ * remain in effect. Iterating this cursor will cause new queries to be sent to the server, even
+ * if the resultant data has already been retrieved by this cursor.
+ *
+ * @return {AggregationCursor} this
+ * @api public
+ * @method rewind
+ */
+
+ QueryCursor.prototype.rewind = function() {
+  this.cursor.rewind();
```

Comment:
Wrap this in `_waitForCursor()`, because `this.cursor` can be undefined until async pre hooks run. Look at addCursorFlag for an example